### PR TITLE
add watch api to provider and region services

### PR DESCRIPTION
**Pull request by aborilov**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apiserver/internal/v1/provider.go [hunk 1]
```diff
@@ -21,34 +21,53 @@ import (
...
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
```
Comment (jiachengxu) [inline pkg/apiserver/internal/v1/provider.go:31]: can you please remove this empty line and resort imports?

File: pkg/apiserver/internal/v1/region.go [hunk 2]
```diff
@@ -21,34 +21,53 @@ import (
...
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
```
Comment (jiachengxu) [inline pkg/apiserver/internal/v1/region.go:31]: can you please remove this empty line and resort imports?

File: pkg/apiserver/internal/v1/provider.go (hunk 1 above)
Comment (aborilov) [reply pkg/apiserver/internal/v1/provider.go:33]: done

File: pkg/apiserver/internal/v1/region.go (hunk 2 above)
Comment (aborilov) [reply pkg/apiserver/internal/v1/region.go:33]: done

**Pull request merged**